gcc-11.3.0/gcc/c-family/c-opts.c

```diff
@@ -1280,6 +1322,39 @@ c_common_finish (void)
 	}
     }
 
+  /* If the calculation of the GitBOM information is enabled, do it here.  */
```

Comment:
Consider adding a comment like:
/* Determine directory to store gitbom files in this order of precedence.
    1. If GITBOM_DIR environment variable is set, use this location.
    2. Use the directory name passed with frecord-gitbom option.
    3. Default is to write the bom files in the same directory as the object file.
   */